modules/images/webp-uploads/load.php

```diff
@@ -554,11 +554,31 @@ function webp_uploads_img_tag_update_mime_type( $image, $context, $attachment_id
 		return $image;
 	}
 
-	$urls = $matches[0];
-	// TODO: Add a filterable option to change the selected mime type. See https://github.com/WordPress/performance/issues/187.
-	$target_mime = 'image/webp';
+	/**
+	 * Filters mime types that should be used to update all images in the content. The order of
+	 * mime types matters. The last mime type in the list will be used if it is supported by an image.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array The list of mime types that can be used to update images in the content.
+	 */
+	$target_mimes = apply_filters( 'webp_uploads_content_image_mimes', array( 'image/jpeg', 'image/webp' ) );
+
+	$target_mime  = null;
+	$target_mimes = array_reverse( $target_mimes );
+	foreach ( $target_mimes as $mime ) {
+		if ( isset( $metadata['sources'][ $mime ] ) ) {
+			$target_mime = $mime;
+			break;
+		}
+	}
+
+	if ( null === $target_mime ) {
+		return $image;
+	}
```

Comment:
We also need to add a condition to short-circuit somehow if the target MIME is the same like the extensions we're replacing (i.e. JPEG).

We can later add another filter to adjust those extensions as well.

Reply:
@felixarntz we already have it down the road: https://github.com/WordPress/performance/blob/f5465574d9640cdb5694f61ecf75a3d97c15b7f1/modules/images/webp-uploads/load.php#L575-L579